CMakeLists.txt

```diff
@@ -409,17 +417,35 @@ if (VAST_ENABLE_STATIC_EXECUTABLE AND BUILD_SHARED_LIBS)
   message(FATAL_ERROR "Cannot create static binary with dynamic libraries")
 endif ()
 
-if (VAST_ENABLE_RELOCATABLE_INSTALLATIONS AND NOT VAST_ENABLE_STATIC_EXECUTABLE)
+if (VAST_ENABLE_RELOCATABLE_INSTALLATIONS)
   # Note that we cannot use the target property INSTALL_RPATH on
   # libvast_internal, because it is an interface library. We should probably fix
   # this by making the config header part of libvast_internal so it is not just
   # an interface library.
-  if (APPLE)
-    list(PREPEND CMAKE_INSTALL_RPATH
-         "@executable_path/../${CMAKE_INSTALL_LIBDIR}")
+  if (NOT VAST_ENABLE_STATIC_EXECUTABLE)
+    if (APPLE)
+      list(PREPEND CMAKE_INSTALL_RPATH
+          "@executable_path/../${CMAKE_INSTALL_LIBDIR}")
+    else ()
+      list(PREPEND CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
+    endif ()
+  endif ()
+
+  # For relocatable builds we need to find the relative difference between the
+  # install prefix and the library location, or the binary location for static
+  # VAST binaries.
+  unset(VAST_LIBDIR_TO_PREFIX)
+  if (VAST_ENABLE_STATIC_EXECUTABLE)
+    set(_libdir "${CMAKE_INSTALL_FULL_BINDIR}")
   else ()
-    list(PREPEND CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
+    set(_libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
   endif ()
+  while (NOT "${_libdir}" STREQUAL "${CMAKE_INSTALL_PREFIX}")
+    get_filename_component(_libdir "${_libdir}" DIRECTORY)
+    list(APPEND VAST_LIBDIR_TO_PREFIX "..")
+  endwhile()
+  unset(_libdir)
+  list(JOIN VAST_LIBDIR_TO_PREFIX "/" VAST_LIBDIR_TO_PREFIX)
```

Comment:
> suspect that this correction will break installed locations, but I don't even get that far:

You're likely using the generated `install` target, which always tries to install the `Release` configuration build, which you likely did not build.

Use `cmake --install path/to/build --prefix path/to/prefix --config <configuration>` to install other configurations.

Reply:
I actually have the same problem when I use `cmake --install`.

Reply:
I just tried it again and it works now, although I didn't do anything different.

You could still apply this patch as a small improvement:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6a16d81e5..c83b17103 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -434,18 +434,16 @@ if (VAST_ENABLE_RELOCATABLE_INSTALLATIONS)
   # For relocatable builds we need to find the relative difference between the
   # install prefix and the library location, or the binary location for static
   # VAST binaries.
-  unset(VAST_LIBDIR_TO_PREFIX)
   if (VAST_ENABLE_STATIC_EXECUTABLE)
     set(_libdir "${CMAKE_INSTALL_FULL_BINDIR}")
   else ()
     set(_libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
   endif ()
-  while (NOT "${_libdir}" STREQUAL "${CMAKE_INSTALL_PREFIX}")
-    get_filename_component(_libdir "${_libdir}" DIRECTORY)
-    list(APPEND VAST_LIBDIR_TO_PREFIX "..")
-  endwhile()
-  unset(_libdir)
   list(JOIN VAST_LIBDIR_TO_PREFIX "/" VAST_LIBDIR_TO_PREFIX)
+  file(RELATIVE_PATH VAST_LIBDIR_TO_PREFIX
+         "${_libdir}"
+         "${CMAKE_INSTALL_PREFIX}")
+  unset(_libdir)
 endif ()

 # -- jemalloc ------------------------------------------------------------------
```

Otherwise lgtm.

Reply:
Will apply the patch. This also needs a rather nasty rebase after #1703.

Reply:
I had to edit it again, now it is with cmake-format applied.